Shared/Extensions/WKWebViewExtensions.swift

```diff
@@ -0,0 +1,97 @@
+    case invalid
+}
+
+extension WKUserScript {
+    public class func createInDefaultContentWorld(source: String, injectionTime: WKUserScriptInjectionTime, forMainFrameOnly: Bool) -> WKUserScript {
+        if #available(iOS 14.0, *) {
+            return WKUserScript(source: source, injectionTime: injectionTime, forMainFrameOnly: forMainFrameOnly, in: .defaultClient)
+        } else {
+            return WKUserScript(source: source, injectionTime: injectionTime, forMainFrameOnly: forMainFrameOnly)
+        }
+    }
+}
+
+public extension WKWebView {
+    func generateJavascriptFunctionString(functionName: String, args: [Any], escapeArgs: Bool = true) -> (javascript: String, error: Error?) {
+        let context = JSContext()
+
+        var sanitizedArgs: [String] = []
+        var error: Error?
+
+        args.forEach {
+            if !escapeArgs {
+                sanitizedArgs.append("\($0)")
+                return
+            }
+            context?.exceptionHandler = { context, exception in
+                if exception != nil {
+                    error = JavascriptError.invalid
+                }
+            }
+            context?.evaluateScript("JSON.parse('\"\($0)\"')")
+            sanitizedArgs.append("'\(String(describing: $0).escapeHTML())'")
+            return
+        }
+        
+        return ("\(functionName)(\(sanitizedArgs.joined(separator: ", ")))", error)
+    }
+
+    func evaluateSafeJavascript(functionName: String, args: [Any] = [], sandboxed: Bool = true, escapeArgs: Bool = true, asFunction: Bool = true, completion: ((Any?, Error?) -> Void)? = nil) {
+        var javascript = functionName
+        
+        if asFunction {
+            let ret = generateJavascriptFunctionString(functionName: functionName, args: args, escapeArgs: escapeArgs)
+            if ret.error != nil {
+                if let completionHandler = completion {
+                    completionHandler(nil, ret.1)
+                }
+                return
+            }
+            javascript = ret.javascript
+        }
+        if #available(iOS 14.0, *), sandboxed {
+            evaluateJavaScript(javascript, in: nil, in: .defaultClient) { result  in
+                switch result {
+                    case .success(let value):
+                        if let completionHandler = completion {
+                            completionHandler(value, nil)
+                        }
+                    case .failure(let error):
+                        if let completionHandler = completion {
+                            completionHandler(nil, error)
+                        }
+                }
+            }
+        } else {
+            evaluateJavaScript(javascript) { data, error  in
+                if let completionHandler = completion {
+                    completionHandler(data, error)
+                }
+            }
+        }
+    }
+}
+
+extension String {
+    /// Encode HTMLStrings
+    func escapeHTML() -> String {
```

Comment:
Lets rename this to `encodingHTMLEntities`

Reply:
Also: If this is not used anywhere else, lets make this `fileprivate`

Reply:
done!